src/edu/tutorials/trainreservation/Main.java

```diff
@@ -23,25 +24,35 @@ public static void main(String[] args) {
 
         Scanner sc = new Scanner(System.in);
         while(true) {
-            System.out.println("\nEnter train search request (type 'exit' to quit)");
+            System.out.println("\nEnter train search request, PNR number or 'REPORT' (type 'exit' to quit)");
 
             String input = sc.nextLine();
             if(input.equals("exit")) {
                 break;
             }
 
-            // Search Trains
-            TrainSearchRequest trainSearchRequest = TrainSearchRequestReader.read(input);
-            System.out.println(trainSearchRequest + "\n");
-
-            List<Train> trainsForRoute = trainService.findTrains(trainSearchRequest);
-            displayTrainNo(trainsForRoute);
-
-            if(!trainsForRoute.isEmpty()) {
-                // Select Train Number
-                String trainNo = DataReader.readTrainNoToBookTicket();
-                Ticket bookedTicket = ticketBookingService.bookTicket(trainNo, trainSearchRequest.getCoachType(), trainSearchRequest.getTravelDate(), trainSearchRequest.getPassengerCount());
-                System.out.println("Ticket booked successfully: " + bookedTicket);
+            if(input.matches("\\d+")) { // if input is a PNR number
+                long pnr = Long.parseLong(input);
+                Ticket ticket = ticketBookingService.getBookingDetails(pnr);
+                if(ticket != null) {
+                    System.out.println(ticket.getTrainNumber() + " " + ticket.getFrom() + " " + ticket.getTo() + " " + ticket.getTravelDate() + " " + ticket.getTotalFare() + " " + Utils.toCommaSeparatedSeatNo(ticket.getBookedSeats()));
+                } else {
+                    System.out.println("Invalid PNR");
+                }
```

Comment:
To make code more readable, we can move this piece of code to a new method "processPNR"